fastlane/lanes/localization.rb

```diff
@@ -223,50 +266,80 @@
-    FileUtils.cp(release_notes_source, File.join(metadata_directory, 'en-US', 'release_notes.txt'))
-
-    metadata_path = File.join(metadata_directory, '**', '*.txt')
-    git_add(path: metadata_path, shell_escape: true)
-    git_commit(
-      path: metadata_path,
-      message: 'Update WordPress metadata translations',
-      allow_nothing_to_commit: true
+    # In the meantime, just copy the file to the right place for `deliver` to find, for the `default` pseudo-locale which is used as fallback
+    release_notes_source = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Jetpack', 'Resources', 'release_notes.txt')
+    FileUtils.cp(release_notes_source, File.join(metadata_directory, 'default', 'release_notes.txt'))
+
+    # Download metadata translations from GlotPress
+    download_localized_app_store_metadata(
+      glotpress_project_url: GLOTPRESS_WORDPRESS_METADATA_PROJECT_URL,
+      metadata_directory: metadata_directory,
+      locales: WORDPRESS_METADATA_GLOTPRESS_LOCALE_CODES,
+      commit_message: 'Update WordPress metadata translations'
     )
   end
 
   # Downloads the localized metadata (for App Store Connect) from GlotPress for the Jetpack app
   #
   desc 'Downloads the localized metadata (for App Store Connect) from GlotPress for the Jetpack app'
   lane :download_jetpack_localized_app_store_metadata do
-    # @FIXME: Replace this whole lane with a call to the future replacement of `gp_downloadmetadata` once it's implemented in the release-toolkit (see paaHJt-31O-p2).
-
-    # No need to `cd` into `fastlane` because of how Fastlane manages its paths internally.
-    sh './download_metadata.swift jetpack'
+    metadata_directory = File.join(PROJECT_ROOT_FOLDER, 'fastlane', 'jetpack_metadata')
 
-    # @TODO: Make the `fastlane/jetpack_metadata/en-US/release_notes.txt` path be the source of truth for the original copies in the future.
+    # FIXME: We should make the `fastlane/jetpack_metadata/default/release_notes.txt` path be the source of truth for the original copies in the future.
     # (will require changes in the `update_appstore_strings` lane, the Release Scenario, the MC tool to generate the announcement post…)
     #
-    # In the meantime, since GlotPress doesn't have the `en-US` notes because those are the ones used as originals, just copy the file to the right place for `deliver` to find
-    metadata_directory = File.join(PROJECT_ROOT_FOLDER, 'fastlane', 'jetpack_metadata')
-    release_notes_source = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Jetpack', 'Resources', 'release_notes.txt')
-    FileUtils.cp(release_notes_source, File.join(metadata_directory, 'en-US', 'release_notes.txt'))
+    # In the meantime, just copy the file to the right place for `deliver` to find, for the `default` pseudo-locale which is used as fallback
+    release_notes_source = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Resources', 'release_notes.txt')
```

Comment:
I think this should be 

```suggestion
    release_notes_source = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Jetpack', 'Resources', 'release_notes.txt')
```

It's easy to miss because for the past few releases we had the same notes between the two apps, but there's always a chance they'll diverge in a future release.

Reply:
Ah! The funny thing is that:
- One of the reasons I intentionally put that code in each of the lanes (as opposed to putting this logic in the `def download_localized_appstore_metadata` helper) was exactly because the paths for the source file were different in the 2 cases… but then I apparently did one too many copy/paste and forgot to finally make them distinct 🤦 
 - I had in mind to add a specific step in the Testing Instructions to mention "Edit the `release_notes.txt` files mentioned at the path in the Release Scenario to provide different copies for the WP and JP files before running the lane, then check the right ones were copied in `default` subfolder of the right metadata folders"… but then I forgot about it when actually writing the PR description 😅

So nice one catching this one despite it being sneaky here 😀👏👍

Reply:
Actually just realized that I _did_ use different `release_notes_source` for WP and JP. It's just that… I swapped them 😓  So I should also apply [this other fix on the other side](https://github.com/wordpress-mobile/WordPress-iOS/pull/18439#discussion_r858167044) 😅